File: LeetCode-Solved-Problems/1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:// neetcode amazing solution
    int minimumDeletions(string s) {
        int a_count_right=0;
        for(char c:s)
            a_count_right += c=='a';
        
        int b_count_left = 0;
        int ans = INT_MAX;
        for(char c:s){
            if(c=='a') a_count_right --;
            
            ans = min(ans, a_count_right + b_count_left);
            
            if(c=='b') b_count_left ++;
        }
        
        return ans;
    }
};

File: LeetCode-Solved-Problems/1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        cost = cnt_b = 0
        for ch in s:
            if ch == 'a' and cnt_b>0:
                cnt_b-=1
                cost+=1
            elif ch == 'b':
                cnt_b+=1
                
        return cost